contracts/Obligation.sol

```diff
@@ -244,16 +244,13 @@ contract Obligation is IObligation, IERC20, European, Ownable, WriterRegistry {
 
         bytes32 id = keccak256(
             abi.encodePacked(
-                expiryTime,
-                windowSize,
-                strikePrice,
-                buyer,
-                seller,
-                // append nonce to avoid
-                // id collisions
+                // create2 ensures params are unique
```

Comment:
@nud3l I don't think we need a global security nonce anymore because the address should be unique and thus prevent collisions across options

Reply:
also we don't want to open ourselves up to spamming attacks if anyone can increment the global nonce

Reply:
Summarizing, we use the unique address of the obligation plus an obligation wide unique nonce to generate an id via keccak256. We are using the full keccak256 in the OP_RETURN. In my opinion that is is sufficient to have a unique hash. It also allows to easier manage the nonce because with a global nonce across all contracts, we would have to create a "filter" for the every growing list of all the contracts that are allowed to access it.